### DISCUSSION TITLE
Using $refs in zodios-openapi

### DISCUSSION THREAD

**Discussion by epotter2297** [ecyrbe/zodios#265] [Ideas]
I see in [zodios-openapi](https://github.com/ecyrbe/zodios-openapi/blob/main/src/openapi.ts#:~:text=%7D-,function%20makeJsonSchema(schema%3A%20z.ZodTypeAny)%20%7B,%7D,-/**) that `$refStrategy: "none"` is set when creating openapi definitions from our zod schemas. I was curious, is it possible for this to work using `root` instead?

The reason I ask is because I have some recursive schemas defined with zod that I would like to show up properly in openapi, but that can only work via a $ref. It looks like `zod-to-json-schema` greatly improved their recursive schema support in their recent `3.20`  version, so that half of the problem is already solved for us 🙂

If that can't work for whatever reason, the same end result could be achieved if I had a way of telling `openApiBuilder` what my recursive schemas are.

Maybe something like:
```ts
const document = openApiBuilder({
  title: "User API",
  version: "1.0.0",
  description: "A simple user API",
})
  .addServer({ url: "/api/v1" })
  .addSecurityScheme("admin", bearerAuthScheme())
  .addPublicApi(api)
  .addRefDefinition(myRecursiveZodSchema)
```
Which would then add `myRecursiveZodSchema` as a ref, and also allow it to be passed in as a `definition` to the `makeJsonSchema()` function in zodios-openapi.

**Comment by ecyrbe** [comment]
Yes, i'll try to activate it and check if the issues are now fixed with zodios

**Reply by lennerd** [reply]
@ecyrbe Any success?

**Comment by lennerd** [comment]
@epotter2297 Any way you were able to get reference definitions with `Zodios`?

**Reply by ecyrbe** [reply]
It was not successfull, needs a lot of custom code to extract the ref, parse it modify it and reinject it in definitions.
If someone has time to do it, help is welcomed

**Reply by lennerd** [reply]
Thanks for trying! 🙏 We solved it for now by going the other way around using OpenAPI to generate a Zodios client. 🤷‍♂️
